[BUG] deprecation warning on `rmm::mr::device_memory_resource::supports_streams`

This is something I saw in my local build of spark-rapids-jni and think we should adjust `SparkResourceAdaptorJni` to use the new `async_memory_resource` concept:

```
INFO]      [exec] [15/36] Building CXX object CMakeFiles/spark_rapids_jni.dir/src/SparkResourceAdaptorJni.cpp.o
[INFO]      [exec] spark-rapids-jni/src/main/cpp/src/SparkResourceAdaptorJni.cpp: In member function 'virtual bool {anonymous}::spark_resource_adaptor::supports_streams() const':
[INFO]      [exec] spark-rapids-jni/src/main/cpp/src/SparkResourceAdaptorJni.cpp:404:86: warning: 'virtual bool rmm::mr::device_memory_resource::supports_streams() const' is deprecated: Functionality removed in favor of cuda::mr::async_memory_resource. [-Wdeprecated-declarations]
[INFO]      [exec]   404 |   bool supports_streams() const noexcept override { return resource->supports_streams(); }
[INFO]      [exec]       |                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~^~
[INFO]      [exec] In file included from spark-rapids-jni/src/main/cpp/src/SparkResourceAdaptorJni.cpp:17:
[INFO]      [exec] spark-rapids-jni/target/libcudf-install/include/rmm/mr/device/device_memory_resource.hpp:306:3: note: declared here
[INFO]      [exec]   306 |   supports_streams() const noexcept
[INFO]      [exec]       |   ^~~~~~~~~~~~~~~~
```

--- a/src/main/cpp/src/SparkResourceAdaptorJni.cpp
+++ b/src/main/cpp/src/SparkResourceAdaptorJni.cpp
@@ -401,8 +401,6 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
 
   rmm::mr::device_memory_resource* get_wrapped_resource() { return resource; }
 
-  bool supports_streams() const noexcept override { return resource->supports_streams(); }
-
   /**
    * Update the internal state so that a specific thread is dedicated to a task.
    * This may be called multiple times for a given thread and if the thread is already

